nms/app/fbcnms-projects/magmalte/app/views/events/EventsTable.js

```diff
@@ -80,17 +78,71 @@ export const magmaEventTypes = Object.freeze({
   SUBSCRIBER: 3,
 });
 
-export default function EventsTable({
+type EventRowType = {
+  ts: string,
+  eventType: string,
+  eventDescription: string,
+  value: {},
+  hardwareID: string,
+  tag: string,
+};
+
+type EventDescriptionProps = {
+  rowData: EventRowType,
+};
+
+function ExpandEvent(props: EventDescriptionProps) {
+  const classes = useStyles();
+  const eventDetails = {
+    hardware_id: props.rowData.hardwareID,
+    tag: props.rowData.tag,
+  };
+  const [expanded, setExpanded] = useState(false);
+  if (props.rowData.value) {
+    for (const [key, value] of Object.entries(props.rowData.value)) {
+      eventDetails[key] = value;
+    }
+  }
+  return (
+    <ExpansionPanel
+      elevation={0}
+      expanded={expanded}
+      onChange={() => setExpanded(!expanded)}>
+      <ExpansionPanelSummary
+        expandIcon={<ExpandMoreIcon />}
+        aria-controls="panel1bh-content">
+        {Object.keys(eventDetails).join(', ')}
+      </ExpansionPanelSummary>
+      <ExpansionPanelDetails>
+        <table>
+          <tbody>
+            {Object.entries(eventDetails).map((entry, i) => (
+              <tr key={i}>
+                <td>{entry[0]}: </td>
+                <td className={classes.eventDetailValue}>
+                  {JSON.stringify(entry[1])}
+                </td>
+              </tr>
+            ))}
+          </tbody>
+        </table>
+      </ExpansionPanelDetails>
+    </ExpansionPanel>
+  );
+}
+
+export default function EventsTable2({
```

Comment:
Does `EventsTable` still exist as a component that's used somewhere? Should we just migrate all usages to this component and rename it `EventsTable`?

If we need both components give this a more appropriate name

Reply:
No EventsTable doesn't exist. Since it is exported as default i guess it was imported as EventsTable everywhere without errors. This was left over by mistake. I will rename this to EventsTable. Thanks for point this out.